diff --git a/schrift.c b/schrift.c
index eb39376..bca3e69 100644
--- a/schrift.c
+++ b/schrift.c
@@ -70,7 +70,7 @@
 /* Allocate values on the stack if they are small enough, else spill to heap. */
 #define STACK_ALLOC(var, type, thresh, count) \
 	type var##_stack_[thresh]; \
-	var = (count) <= (thresh) ? var##_stack_ : calloc(sizeof(type), count);
+	var = (count) <= (thresh) ? var##_stack_ : calloc(count, sizeof(type));
 #define STACK_FREE(var) \
 	if (var != var##_stack_) free(var);
 
